packages/mason_cli/lib/src/commands/search.dart

```diff
@@ -24,29 +32,75 @@ class SearchCommand extends MasonCommand {
     if (results.rest.isEmpty) usageException('search term is required.');
 
     final query = results.rest.first;
+    final isGlobalSearch = results['global'] == true;
     final searchProgress = logger.progress(
-      'Searching "$query" on brickhub.dev',
+      isGlobalSearch
+          ? 'Searching "$query" globally'
+          : 'Searching "$query" on brickhub.dev',
     );
 
     try {
-      final results = await _masonApi.search(query: query);
-      searchProgress.complete(
-        results.isEmpty
-            ? 'No bricks found.'
-            : '''Found ${results.length} brick${results.length == 1 ? '' : 's'}.''',
-      );
-
-      logger.info('');
-
-      for (final brick in results) {
-        logger
-          ..info(
-            lightCyan.wrap(styleBold.wrap('${brick.name} v${brick.version}')),
-          )
-          ..info(brick.description)
-          ..info('https://brickhub.dev/bricks/${brick.name}/${brick.version}')
-          ..info(darkGray.wrap('-' * 80));
+      if (isGlobalSearch) {
+        final bricks = globalBricks;
+
+        logger.info(path.dirname(globalMasonYamlFile.path));
+
+        if (bricks.isEmpty) {
+          logger.info('└── (empty)');
+          return ExitCode.success.code;
+        }
+
+        final sortedBricks = [...bricks]..sort(
+            (a, b) => a.name.compareTo(
+              b.name,
+            ),
+          );
+
+        final searchResults = sortedBricks.where(
+          (brick) {
+            final contains = brick.name.contains(query);
+            return contains;
+          },
+        ).toList();
+
+        searchProgress.complete(
+          searchResults.isEmpty
+              ? 'No bricks found.'
+              : '''Found ${searchResults.length} brick${searchResults.length == 1 ? '' : 's'}.''',
+        );
+
+        final responce = logger.chooseOne(
+          'Search results',
+          choices: searchResults.map<String>((e) {
+            return '${e.name} ${e.version} -> ${e.path}';
+          }).toList(),
+        );
+        final name = responce.trim().split(' ')[0];
+
+        logger.info('Selected $name brick');
+
+        // TODO: Need to find a better way to call brick generation command.
```

Comment:
This is brick generation step after selection of brick from search results. 
I think this is not the best way to call the brick generation command, so looking for some advice on how I can call this command safely and more properly.